src/Common/Infrastructure/Extensions/AcceptTypeExtensions.cs

```diff
@@ -108,10 +110,31 @@ public static string ToProblemResponseMimeTypeString(this AcceptType acceptType)
                 }
             }
 
-            throw new ApiException("Ongeldig formaat.", StatusCodes.Status406NotAcceptable);
+            throw InvalidAcceptType;
         }
 
-        private static bool Contains(this MediaTypeHeaderValue headerValue, string mineType)
-            => headerValue.MediaType.Value.Contains(mineType, StringComparison.InvariantCultureIgnoreCase);
+        public static AcceptType ValidateFor(
+            this AcceptType acceptType,
+            EndpointType endpointType)
+            => endpointType
+                .GetAcceptedTypes()
+                .Contains(acceptType)
+                ? acceptType
+                : throw InvalidAcceptType;
+
+        public static bool Contains(
+            this RequestHeaders requestHeaders,
+            AcceptType acceptType)
+            => requestHeaders
+                .Accept
+                .Any(headerValue => headerValue.Contains(acceptType.ToMimeTypeString()));
```

Comment:
Should we take `*/*` into account as an Accept header? Since that implies "anything is fine"

Reply:
`*/*` is already converted to an `AcceptType (enum)` at that point

Reply:
Ah :+1:

Reply:
but could add it for being complete, in case the check would be used on other places